src/ddmd/mars.d

```diff
@@ -245,6 +245,29 @@ extern (C++) void genCmain(Scope* sc)
     rootHasMain = sc._module;
 }
 
+private void semantic3OnDependencies(Module m, StringTable *importArray)
+{
+    if(!m)
+        return;
+
+    const(char) *mName = m.toChars();
+    size_t mNameLen    = strlen(mName);
+
+    if (importArray.lookup(mName, mNameLen))
+        return;
+
+    if (global.params.verbose)
+        fprintf(global.stdmsg, "semantic3 %s\n", mName);
+
+    importArray.insert(mName, mNameLen, cast(void*) mName);
+    m.semantic3(null);
+
+    for (int i=1; i<m.aimports.dim; i++)
```

Comment:
Actually, another issue is that this is using `int` for the index when it should be `size_t`.